### PR TITLE
Create dotnet base image and add documentation

### PR DESCRIPTION
**Pull request by codegold79**
Signed-off-by: codegold79 <codegold79@gmail.com>

### CODE REVIEW

File: dotnet/README.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Containerize a dotnet project
+
+This is the incomplete beginning of a dotnet image created by following [Microsoft Docs](https://docs.microsoft.com/en-us/dotnet/core/install/linux-package-manager-debian10). Projects that use dotnet and ASP.net should use this image as their base.
```
Comment (oliverdain) [inline dotnet/README.md]: Please wrap lines in READMEs to 120 characters - they're read via `less` from the terminal more often than they're read in github.

File: dotnet/README.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+This base image is available at mvpstudio/dotnet:<VERSION>. The Dockerfile is placed in the directory with the base image version designation.
+
+As for which versions of dotnet runtime, aspnetcore runtime, and dotnet sdk are being used, take a look at the `Dockerfile` which versions inside one of the version directories. The v1 base image uses the 3.1 dotnet and aspnetcore runtimes.
+
+To create a complete a Dockerfile for individual projects, be sure to create a Dockerfile with the content shown below. Replace <VERSION> with the latest version, e.g. `v1`, and all instances of `LittleHelpBook` with the appropriate project name. Once these replacements are made, the Dockerfile should be complete.
+
+```Docker
+FROM mvpstudio/dotnet:<VERSION> as base
+FROM mcr.microsoft.com/dotnet/core/sdk:3.1-buster AS build
```
Comment (oliverdain) [inline dotnet/README.md:21]: Nice! I like that you have minimal runtime stuff in `mvpstudio/dotnet` and then use the MS tools to add in just what you need to build it.

File: dotnet/README.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
+# Containerize a dotnet project
+
+This is the incomplete beginning of a dotnet image created by following [Microsoft Docs](https://docs.microsoft.com/en-us/dotnet/core/install/linux-package-manager-debian10). Projects that use dotnet and ASP.net should use this image as their base.
+
+This base image is available at mvpstudio/dotnet:<VERSION>. The Dockerfile is placed in the directory with the base image version designation.
+
+As for which versions of dotnet runtime, aspnetcore runtime, and dotnet sdk are being used, take a look at the `Dockerfile` which versions inside one of the version directories. The v1 base image uses the 3.1 dotnet and aspnetcore runtimes.
+
+To create a complete a Dockerfile for individual projects, be sure to create a Dockerfile with the content shown below. Replace <VERSION> with the latest version, e.g. `v1`, and all instances of `LittleHelpBook` with the appropriate project name. Once these replacements are made, the Dockerfile should be complete.
```
Comment (oliverdain) [inline dotnet/README.md]: > with the content shown below.

That content is, in places, specific to little help book (e.g. lines 15 and 16) so I'd just say "content similar to" or something.

File: dotnet/v1/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,17 @@
+FROM mvpstudio/base:v1 AS base
+ENV ASPNETCORE_URLS=http://*:8080
+
+WORKDIR /home/mvp/app
+EXPOSE 8080
+
+# https://docs.microsoft.com/en-us/dotnet/core/install/linux-package-manager-debian10
+RUN apt-get install wget && \
```
Comment (oliverdain) [inline dotnet/v1/Dockerfile:8]: total nit-pick but it's semi-standard to indent follow-up lines when you have long multi-line things like this. Doesn't make any "real" difference - I think most people just find that easier to read.

File: dotnet/v1/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+wget -O- https://packages.microsoft.com/keys/microsoft.asc | \
+gpg --dearmor > microsoft.asc.gpg && \
+mv microsoft.asc.gpg /etc/apt/trusted.gpg.d/ && \
+wget https://packages.microsoft.com/config/debian/10/prod.list && \
+mv prod.list /etc/apt/sources.list.d/microsoft-prod.list && \
+chown root:root /etc/apt/trusted.gpg.d/microsoft.asc.gpg && \
+chown root:root /etc/apt/sources.list.d/microsoft-prod.list && \
```
Comment (oliverdain) [inline dotnet/v1/Dockerfile]: Are these `chown` commands necessary? Is't this running as root so they be owned by root by default?

File: dotnet/v1/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+apt-get update && echo "Y" | apt-get install dotnet-runtime-3.1 && \
```
Comment (oliverdain) [inline dotnet/v1/Dockerfile]: `apt-get install -y` is very similar to `echo "Y" | apt-get install` but answers `Y` to all questions asked, even if there's more than 1. Also shorter.

**Review by oliverdain — APPROVED** [review]
Nice!

File: dotnet/v1/Dockerfile (hunk 5 above)
Comment (codegold79) [reply dotnet/v1/Dockerfile]: Good point--I'll have them removed and test it.

File: dotnet/README.md (hunk 1 above)
Comment (codegold79) [reply dotnet/README.md]: I've wrapped them to around 100 chars

**Comment by oliverdain** [comment]
@codegold79 I gave you an approve which means you're free to merge after making the requested changes.

**Pull request merged**